pkg/resources/validation/cluster/cluster.go

```diff
@@ -93,16 +154,136 @@ func (v *Validator) Admit(request *admission.Request) (*admissionv1.AdmissionRes
+		return psa.AdmissionResponseBadRequest(msg), nil
+	}
+	if newTemplateName != "" {
+		response, err := v.checkPSAConfigOnCluster(newCluster)
+		if err != nil {
+			return nil, fmt.Errorf("failed to check the PodSecurity Config in the cluster %s: %w", newCluster.Name, err)
+		}
+		if !response.Allowed {
+			return response, nil
+		}
+	} else {
+		switch request.Operation {
+		case admissionv1.Create:
+			return psa.AdmissionResponseAllowed(), nil
+		case admissionv1.Update:
+			// In the case of unsetting DefaultPodSecurityAdmissionConfigurationTemplateName,
+			// validate that the configuration for PodSecurityAdmission under the kube-api.admission_configuration section
+			// is different between the new and old clusters.
+			// It is possible that user drops DefaultPodSecurityAdmissionConfigurationTemplateName and set the config
+			// under kube-api.admission_configuration at the same time.
+			if oldTemplateName != "" {
+				newConfig, found := psa.GetPluginConfigFromCluster(newCluster)
+				if !found {
+					// not found means the kube-api.admission_configuration section is also removed, which is good
+					return psa.AdmissionResponseAllowed(), nil
+				}
+				oldConfig, _ := psa.GetPluginConfigFromCluster(oldCluster)
+				if reflect.DeepEqual(newConfig, oldConfig) {
+					msg := "The Plugin Config for PodSecurity under kube-api.admission_configuration is the same as the previously-set PodSecurityAdmissionConfigurationTemplate." +
+						" Please either change the Plugin Config or set the DefaultPodSecurityAdmissionConfigurationTemplateName."
+					return psa.AdmissionResponseBadRequest(msg), nil
+				}
+			}
+		}
+	}
+	return psa.AdmissionResponseAllowed(), nil
 }
 
-func toExtra(extra map[string]authenticationv1.ExtraValue) map[string]v1.ExtraValue {
-	result := map[string]v1.ExtraValue{}
-	for k, v := range extra {
-		result[k] = v1.ExtraValue(v)
+// checkPSAConfigOnCluster validates the cluster spec when DefaultPodSecurityAdmissionConfigurationTemplateName is set
+func (v *Validator) checkPSAConfigOnCluster(cluster *apisv3.Cluster) (*admissionv1.AdmissionResponse, error) {
+	// validate that extra_args.admission-control-config-file is not set at the same time
+	_, found := cluster.Spec.RancherKubernetesEngineConfig.Services.KubeAPI.ExtraArgs["admission-control-config-file"]
+	if found {
+		msg := "could not use external admission control configuration file when using PodSecurityAdmissionConfigurationTemplate"
+		return psa.AdmissionResponseBadRequest(msg), nil
 	}
-	return result
+	// validate that the configuration for PodSecurityAdmission under the kube-api.admission_configuration section
+	// matches the content of the PodSecurityAdmissionConfigurationTemplate specified in the cluster
+	name := cluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+	template, err := v.psact.Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
What if PSACT used by this cluster got deleted? This would prevent any updates to the cluster.
Maybe we need to further discuss it, but I thought that after PSACT is specified we basically copy its contents down and that relationship to PSACT is severed and we no longer synchronize updates to PSACT down to all clusters that use it - but maybe I'm wrong here.

Reply:
1. This check confirms that the content under the `kube-api.admission_configuration` matches the specified PSACT; therefore, we need to get the PSACT. 
2. To avoid the event you concerned, we do not allow deletion of a PSACT if any cluster uses it, which has been added to the webhook recently [here](https://github.com/rancher/webhook/blob/2b9e6c6301e489122b034101b9e7ed9ae8f7106e/pkg/resources/validation/podsecurityadmissionconfigurationtemplate/podsecurityact.go#L134-L148)